src/shared/Core/Trace2FileWriter.cs

```diff
@@ -0,0 +1,18 @@
+using System.IO;
+
+namespace GitCredentialManager;
+
+public class Trace2FileWriter : Trace2Writer
+{
+    private readonly string _path;
+
+    public Trace2FileWriter(Trace2FormatTarget formatTarget, string path) : base(formatTarget)
+    {
+        _path = path;
+    }
+
+    public override void Write(Trace2Message message)
+    {
+        File.AppendAllText(_path, $"{Format(message)}\n");
```

Comment:
File.AppendAllText() feels expensive, but I guess that you have to do it to get proper APPEND semantics.  Sigh.

If I understand your Trace2Writer.Format() method, it can fail -- set Failed=true and return "" -- do we need to guard against that here ?   Seems like if it does fail, we'll start writing blank lines to the file/socket.  (or maybe only one)

Reply:
Would it be better to append the LF inside the Format() method?  Then we don't need to deal with it here and do a second $"{}" style printf.

Reply:
Matthew and I wrestled with this for awhile yesterday and came to the conclusion that `File.AppendAllText` is the only way to ensure we're writing correctly 😬.

We won't need to guard, since we ensure we don't call `Write` if the writer has failed [in `Trace2`](https://github.com/git-ecosystem/git-credential-manager/blob/622960273ecf31b2b23d2c93fc68462c7d6cc342/src/shared/Core/Trace2.cs#LL292-L295C18).

I did go ahead and update Format to append the newline - thanks for the pointer!